evm_arithmetization/src/cpu/kernel/interpreter.rs

```diff
@@ -1142,6 +1145,29 @@ impl<'a, F: RichField> Interpreter<'a, F> {
         self.push(hash)
     }
 
+    fn run_poseidon_general(&mut self) -> anyhow::Result<(), ProgramError> {
+        let base_address = MemoryAddress::new_bundle(self.pop()?)?;
+        let len = u256_to_usize(self.pop()?)?;
+
+        let mut input = (0..len)
+            .map(|i| {
+                let address = MemoryAddress {
+                    virt: base_address.virt.saturating_add(i),
+                    ..base_address
+                };
+                let val = self.generation_state.memory.get(address);
+
+                val.0[0] as u8
+            })
+            .collect_vec();
+
+        // TODO: We are assuming the bytes are already padded
```

Comment:
Why is it a TODO? Because that's not actually the case?

Reply:
It was more a note to myself in case it was not the case

Reply:
Sorry, I misread what you wrote. Actually it is the case that the data is already padded 
```assembly
// Load the code non-deterministically in memory and return the length.
  PROVER_INPUT(account_code)
// stack: padded_code_size, codehash, ctx, retdest
  %jump(poseidon_hash_code)`
``` 
cpu/kernel/asm/account_code.asm#L88`